File: app/src/main/java/com/eftimoff/bakingapp/recipelist/view/RecipeAdapter.java
package com.eftimoff.bakingapp.recipelist.view;

import android.support.v7.widget.RecyclerView;

import com.eftimoff.bakingapp.app.models.Recipe;

import java.util.List;

public abstract class RecipeAdapter extends RecyclerView.Adapter<RecipeViewHolder> {

    interface Callback {
        void onRecipeClicked(Recipe recipe);
    }

    abstract void setRecipes(List<Recipe> recipes);

    abstract void setCallback(Callback callback);
    
}

File: app/src/main/java/com/eftimoff/bakingapp/recipelist/view/RecipeAdapter.kt
package com.eftimoff.bakingapp.recipelist.view

import android.support.v7.widget.RecyclerView
import com.eftimoff.bakingapp.app.models.Recipe

abstract class RecipeAdapter : RecyclerView.Adapter<RecipeViewHolder>() {

    interface Callback {
        fun onRecipeClicked(recipe: Recipe)
    }

    abstract fun setRecipes(recipes: List<Recipe>)

    abstract fun setCallback(callback: Callback)

}